### PR TITLE
gar/provenance pr

### PR DESCRIPTION
**Pull request by wraithgar**
- feat: add provenance attestation
- fix: refactor error reporting in audit command

### CODE REVIEW

**Comment by wraithgar** [comment]
This is the actual PR to land for provenance.  We are doing it this way so that the `provenance` branch does not get removed when the PR is landed.

File: lib/commands/audit.js [hunk 1]
```diff
@@ -76,54 +75,66 @@ class VerifySignatures {
...
-      const msg = this.verifiedCount === 1 ?
-        `${this.verifiedCount} package has a ${verifiedBold} registry signature\n` :
-        `${this.verifiedCount} packages have ${verifiedBold} registry signatures\n`
-      this.appendOutput(msg)
+      if (this.verifiedCount === 1) {
+        this.npm.output(`${this.verifiedCount} package has a ${verifiedBold} registry signature`)
+      } else {
+        this.npm.output(`${this.verifiedCount} packages have ${verifiedBold} registry signatures`)
+      }
+      this.npm.output('')
```
Comment (ljharb) [inline lib/commands/audit.js:88]: these seem like unrelated refactors, is it worth landing those in a separate PR to make this diff smaller?

File: test/lib/commands/audit.js [hunk 2]
```diff
@@ -1447,7 +1447,7 @@ t.test('audit signatures', async t => {
 
     await npm.exec('audit', ['signatures'])
 
-    t.equal(process.exitCode, 0, 'should exit successfully')
+    t.notOk(process.exitCode, 'should exit successfully')
```
Comment (ljharb) [inline test/lib/commands/audit.js:1450]: why the less explicit checks?

File: lib/commands/audit.js (hunk 1 above)
Comment (wraithgar) [reply lib/commands/audit.js:88]: the commits are separate

Comment (ljharb) [reply lib/commands/audit.js:88]: true enough, i just looked at the whole diff view :-) that makes it easier to land them separately tho, no?

Comment (wraithgar) [reply lib/commands/audit.js:88]: this branch has been very long lived (`provenance`) and we've been testing and reviewing it for months.  originally the second commit had other work in it that we pulled out for another PR but this was left here because it already existed and is just a refactor. it will be landed as two commits.

File: test/lib/commands/audit.js (hunk 2 above)
Comment (wraithgar) [reply test/lib/commands/audit.js:1450]: process.exitCode is a number this check is equally as explicit.

**Review by bdehamer — APPROVED** [review]
LGTM 🎉
